aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -111,6 +130,91 @@ fn native_get_script_hash(
     )])
 }
 
+fn native_sender_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SENDER_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        Ok(smallvec![Value::address(transaction_context.sender())])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::E_TRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_secondary_signers_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SECONDARY_SIGNERS_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        let secondary_signers = transaction_context.secondary_signers();
+        context.charge(
+            TRANSACTION_CONTEXT_SECONDARY_SIGNERS_PER_SIGNER
+                * NumArgs::new(secondary_signers.len() as u64),
+        )?;
+        Ok(smallvec![Value::vector_address(secondary_signers)])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::E_TRANSACTION_CONTEXT_NOT_AVAILABLE),
```

Comment:
How does custom error message work here?

Reply:
`ETRANSACTION_CONTEXT_NOT_AVAILABLE` is also defined with a proper error message in `transaction_context.move` as follows:
```
/// Transaction context is not available outside of the transaction prologue, execution, or epilogue phases.
const ETRANSACTION_CONTEXT_NOT_AVAILABLE: u64 = 1;
 ```